pkg/multierror/prefixed.go

```diff
@@ -0,0 +1,111 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package multierror
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+// FormatFunc defines a format function which should format a slice of errors
+// into a string.
+type FormatFunc func(es []error) string
+
+// Prefixed is a multierror which will prefix the error output message with the
+// specified prefix.
+type Prefixed struct {
+	Prefix     string
+	Errors     []error
+	FormatFunc FormatFunc
+}
+
+// NewPrefixed creates a new pointer to Prefixed w
+func NewPrefixed(prefix string, errs ...error) *Prefixed {
```

Comment:
It's a bit strange how clients are supposed to use the `Prefixed` struct
`prefix` and `errs` can be passed a vars but then `FormatFunc` should be set with dot notation assignment. 
Is this what we want?

```
merr := NewPrefixed("config validation")
merr.FormatFunc = someFunc()
merr = merr.Append(errs)
```

Reply:
Well, the idea is that there's no need to set the format func, most users won't bother changing that function and just rely on the default one I assume. It can always be set `to a function. I could add a setter method, but since the field is public, all can be modified, and so are the other structure fields, being all public, there's no need to use the constructor either, it can be used, but there's no need really.